src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package kitchenpos.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import kitchenpos.domain.Menu;
+import kitchenpos.domain.MenuGroup;
+import kitchenpos.domain.Product;
+import kitchenpos.repository.MenuGroupRepository;
+import kitchenpos.repository.ProductRepository;
+import kitchenpos.ui.dto.request.MenuCreateRequest;
+import kitchenpos.ui.dto.request.MenuProductRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class MenuServiceTest extends ServiceTest {
+    @Autowired
+    private MenuService menuService;
+    @Autowired
+    private MenuGroupRepository menuGroupRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    private MenuGroup menuGroup;
+    private Product productA;
+    private Product productB;
+    private String name;
+    private BigDecimal price;
+    private Long menuGroupId;
+    private MenuProductRequest menuProductA;
+    private MenuProductRequest menuProductB;
+
+    @BeforeEach
+    void setUpForMenu() {
+        menuGroup = menuGroupRepository.save(new MenuGroup("순살 두 마리"));
+        productA = productRepository.save(new Product("순살 까르보치킨", new BigDecimal(20000)));
+        productB = productRepository.save(new Product("순살 짜장치킨", new BigDecimal(19000)));
+        name = "순살 까르보 한 마리 + 순살 짜장 한 마리";
+        price = new BigDecimal(35000);
+        menuGroupId = menuGroup.getId();
+        menuProductA = new MenuProductRequest(productA.getId(), 1L);
+        menuProductB = new MenuProductRequest(productB.getId(), 1L);
+    }
+
+    @DisplayName("메뉴를 생성할 수 있다")
+    @Test
+    void create() {
+        // given
+        final var menuRequest = new MenuCreateRequest(name, price, menuGroupId, List.of(menuProductA, menuProductB));
+
+        // when
+        final var menu = menuService.create(menuRequest);
+        ;
+
+        // then
+        assertAll(
+                () -> assertThat(menu.getId()).isEqualTo(1L),
+                () -> assertThat(menu.getName()).isEqualTo(name),
+                () -> assertThat(menu.getPrice().doubleValue()).isEqualTo(price.doubleValue()),
+                () -> assertThat(menu.getMenuGroupId()).isEqualTo(menuGroupId),
+                () -> assertThat(menu.getMenuProducts()).extracting("menuId")
+                        .containsExactly(menu.getId(), menu.getId())
+        );
```

Comment:
```suggestion
        // then
        assertAll(
                () -> assertThat(menu)
                        .usingRecursiveComparison()
                        .isEqualTo(new Menu(1L, name, price, menuGroupId, menu.getMenuProducts())),
                () -> assertThat(menu.getMenuProducts()).extracting("menuId")
                        .containsExactly(menu.getId(), menu.getId())
        );
```

현재 서비스 코드에서 order의 필드 데이터를 하나씩 assertThat으로 검증해주고 있는데, 이를 통합해 검증할 만한 메소드를 하나 만들면 어떨까요? 아니면 기존 메소드인 usingRecursiveComparison 을 사용하는 방법도 있어요!

Reply:
저는 테스트 코드에서 추가적인 메서드를 구현하는 방식을 가급적 피하는 걸 선호해요.
테스트에서는 다소 길어지더라도 코드를 바로 드러내는 것이 읽는 입장에서 더 편하다고 생각하는 편입니다만..
개인차가 있을 수 있을 것 같네요.

usingRecursive를 사용하는 제안해주신 코드로 테스트를 돌려봤는데
BigDecimal에서 값이 다르다고 테스트가 통과하지 못하네요ㅠ

지적해주신 부분에 대해 반복되는 코드가 많음을 인지하고 있어서..
BigDecimal 관련 이슈를 해결해서 usingRecursive를 사용하는 방향을 탐색해보겠습니다~!

Reply:
BigDecimal 생성시 매개변수로 소숫점 두자리까지 포함한 String을 전달하는 식으로 개선한 뒤로
usingRecursive가 정상 동작하네요!!

다른 부분에도 적용할 부분을 살펴봤는데
사알짝 애매해서.. 다른 부분에는 남겨뒀습니다!